### ISSUE TITLE
Potions don't get automatically refreshed all the time in multiplayer

### ISSUE THREAD

**Issue by VilcsiTH** [VoidRift/AutoPotion#1]
I've noticed that in multiplayer, once potions run out, they aren't getting reapplied automatically. This only started happening recently, since we started the lunar pillar event. I've talked about it with other players, but they don't seem to have noticed.
I can't reproduce this bug reliably. Restarting the game fixes it for a while, but the bug starts happening again after a random amount of time, and doesn't go away on its own. Disabling and enabling the automatic potion drinking re-applies the missing effects, but after they run out, they aren't applied again.
You can see a few buffs running out on this image:
![01_potion expiring](https://user-images.githubusercontent.com/7689214/181377051-83fe5bd3-cffd-428d-a848-433cab98efd5.jpg)
Afterwards, even though the chat says automatic potion drinking is enabled, the buffs aren't getting reapplied:
![02_expired even though enabled](https://user-images.githubusercontent.com/7689214/181377053-393a56d7-3d22-47aa-b50f-0974e9d8b618.jpg)
Disabling and enabling automatic potion drinking refreshes the missing buffs:
![03_reenabling solves](https://user-images.githubusercontent.com/7689214/181377030-cc9cb4e7-b25f-42dc-95c0-9226dc7e9bc2.jpg)
I've included the client and server options:
![04_client options](https://user-images.githubusercontent.com/7689214/181377041-743e20c1-1d2b-4a48-8669-c8291c45139a.jpg)
![05_server options](https://user-images.githubusercontent.com/7689214/181377045-64c5005d-7bdf-46d8-a18e-9a22ae943e53.jpg)
I've also attached the client.log, but it doesn't say anything, the screenshots above were taken at 23:31. Useful for the loaded mods list, I suppose:
[autopotionbug.client.log](https://github.com/VoidRift/AutoPotion/files/9203738/autopotionbug.client.log)

**Comment by VoidRift** [comment]
I updated the mod with some changes and added some logs. I can't really reproduce the bug on my side. And I checked the logic of my mod and I can't really see how the bug could happen.
Maybe if you can consistently reproduce it you could send me the potions in the inventory.
Or maybe a player + world + mods zip file would be easiest.

**Comment by VilcsiTH** [comment]
Caught it. Seems like something triggers a loop. Thanks for the update, I've included the fresh log.
[potionloop.client.log](https://github.com/VoidRift/AutoPotion/files/9204366/potionloop.client.log)

Only additional potions I use are from Calamity and I have no other potion mods. I do not know which effect could be 575, but I use a potion, "Fabsol's Vodka", that inflicts a debuff rather than a buff. I'm not sure if this could be the cause, but I felt like I should mention it just in case.

**Comment by VoidRift** [comment]
There was a problem where certain potions turn off / disable other potions which would cause my mod to continuously try and consume a new potion. For example: Profaned Rage Potion + Rage Potion. The tool tip says which potions are compatible. I think I made exceptions for all potions which don't work together but there might be others that I missed. 
This might be related but I don't have high hopes.
I updated the mod and if it is still happening telling me with what potions it happens would help immensely.

**Comment by VilcsiTH** [comment]
Sadly, it still happens. I went through my list of potions to make sure I'm not using anything conflicting, and I haven't found anything so far. I even counted how many potions I'm using and they match up with the number of buffs I receive. The log still lists the same error: "Potion with buff: 575 was not in activatedPotion list." It repeats this message 55 times per second, so almost every frame.(I have Frame Skip on Subtle with the framerate limit on 60, but I still dip sometimes because of effects on screen.)

I've compiled a list of my potions and their effects to save some time, aswell as the buffs I receive and the armor and accessories I'm currently using.
[potionsbuffsarmoraccessories.txt](https://github.com/VoidRift/AutoPotion/files/9214141/potionsbuffsarmoraccessories.txt)
Here's a picture to compare them to.
![containerequipmentbuffs](https://user-images.githubusercontent.com/7689214/181635534-bf31b813-3e8a-43b8-b696-526ba8239ee8.jpg)
